crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -170,4 +170,115 @@ def cancel
   def repocheck
     render json: Api::Upgrade.repocheck
   end
+
+  protected
+
+  api :POST, "/api/upgrade/new",
+    "Initialization of Crowbar during upgrade with creation of a new database.
+    NOTE: It is only possible to use this endpoint during the stage where crowbar-init is running."
+  header "Accept", "application/vnd.crowbar.v2.0+json", required: true
+  param :username, /(?=^.{4,63}$)(?=^[a-zA-Z0-9_]*$)/,
+    desc: "Username
+      Min length: 4
+      Max length: 63
+      Only alphanumeric characters or underscores
+      Must begin with a letter [a-zA-Z] or underscore",
+    required: true
+  param :password, /(?=^.{4,63}$)(?=^[a-zA-Z0-9_]*$)(?=[a-zA-Z0-9_$&+,:;=?@#|'<>.^*()%!-]*$)/,
+    desc: "Password
+      Min length: 4
+      Max length: 63
+      Alphanumeric and special characters
```

Comment:
What are considered special characters here?
ä, ö, ü, ë, ñ?
Symbols?
Is there any restriction or everything is supported?